Update pin_memory.py #116403

Yashovardhan08 opened the pull request:
Update pin_memory.py to include specific device, if mentioned

Fixes #116403

Yashovardhan08 committed:
Update pin_memory.py 

Update pin_memory.py to include specific device, if mentioned

vadimkantorov commented on torch/utils/data/_utils/pin_memory.py:
```diff
@@ -74,6 +74,6 @@ def pin_memory(data, device=None):
             return [pin_memory(sample, device) for sample in data]
     elif hasattr(data, "pin_memory"):
-        return data.pin_memory()
+        return data.pin_memory(device)
```
I wonder if some hacks like checking method signature are needed, as probably for many years implementors assumed and implemented an argument-less variant

Yashovardhan08 replied:
Sure, I will try to add those checks

Yashovardhan08 replied:
@vadimkantorov I have added some basic checks, please advise

vadimkantorov replied:
I'm not maintainer :) We need to wait for some real reviewers. The above is just a consideration that seems reasonable. But I don't have a larger design picture in mind

Yashovardhan08 committed:
Update pin_memory.py

Added arguments check to verify arguments and call the function "pin_memory" on the custom python object.

albanD commented:
Change sonuds good to me.
I'll let @jbschlosser guide you through the rest of the process since he marked the issue as actionable.

vadimkantorov commented:
@alband should this check be cached to be not performed at every pinning?

Yashovardhan08 commented:
Will this be reviewed? 
@albanD  @jbschlosser
